spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,377 @@ def set_tabs_test(option)
+
+      it 'When the user specifies both type and checkin but there are only partial matches' do
+        core.cmd_sessions('--search', 'session_id:1 session_type:something')
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context 'searches for checkin with sessions that do not respond to checkin' do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::CommandShell, type: 'cmd_shell', sid:1, sname: 'session1', info: 'info', session_host: '127.0.0.1', tunnel_to_s: 'tunnel'),
+        }
+      end
+
+      it 'When the user searches for checkin values' do
+        core.cmd_sessions('--search', 'last_checkin:after:6s')
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context 'with other flags' do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: 'session1', info: 'info', session_host: '127.0.0.1', tunnel_to_s: 'tunnel'),
+          2 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:2, sname: 'session2', info: 'info', session_host: '127.0.0.1', tunnel_to_s: 'tunnel'),
+          3 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:3, sname: 'session3', info: 'info', session_host: '127.0.0.1', tunnel_to_s: 'tunnel')
+        }
+      end
+      it 'When the user tries to kill all matching sessions but there are no matches' do
+        core.cmd_sessions('--search', 'session_id:5', '-K')
+        expect(@combined_output).to eq([
+          'No matching sessions.'
+        ])
+      end
+
+      it 'When the user tries to kill all matching sessions and there are matches' do
+        expect(sessions[1]).not_to receive(:kill)
+        expect(sessions[2]).to receive(:kill)
+        expect(sessions[3]).to receive(:kill)
+        core.cmd_sessions('--search', 'session_id:2 session_id:3', '-K')
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Killing matching sessions...
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+            3   session3  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+    end
+  end
+
+  describe '#parse_duration' do
+    {
+          '1s' => 1,
+          '2s' => 2,
+          '3.5s' => 3,
+          '1.5m' => 90,
+          '1.5d' => 129600,
+          '1d1h1m1s' => 90061,
+          '1.5d1.5h1.5m1.5s' => 135091,
+          '1.75m70s' => 175
+    }.each do |input, expected|
+      it 'returns #{expected} seconds for the input #{input}' do
```

Comment:
String interpolation only works with double quotes, and the indentation is a bit off above; I think Rubocop might've caught both of these

Original comment:
https://github.com/rapid7/metasploit-framework/pull/18364#discussion_r1357188188